spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesCatalogWatchAutoConfiguration.java

```diff
@@ -45,8 +47,8 @@ public class KubernetesCatalogWatchAutoConfiguration {
 	@ConditionalOnProperty(name = "spring.cloud.kubernetes.discovery.catalog-services-watch.enabled",
 			matchIfMissing = true)
 	public KubernetesCatalogWatch kubernetesCatalogWatch(KubernetesClient client,
-			KubernetesDiscoveryProperties properties) {
-		return new KubernetesCatalogWatch(client, properties);
+			KubernetesDiscoveryProperties properties, Environment environment) {
```

Comment:
this needs your closer look, as this is a breaking change. I was thinking that if we fix an issue (and imho, we do here), this would be OK. But your input is of course needed